### PR TITLE
Updating our paths for intellisense, and support it for ObjectiveC++

### PR DESCRIPTION
**Pull request by woody-apple**
Our paths are out of date, and Objective C++ files weren't using this. Fixing that :)

### CODE REVIEW

**Review by anush-apple — APPROVED** [review]

File: .vscode/settings.json [hunk 1]
```diff
@@ -14,7 +14,18 @@
...
-        "${workspaceFolder}/third_party/nlio/repo/include/**"
+        "${workspaceFolder}/third_party/nlio/repo/include/**",
+        "${workspaceFolder}/darwin/Framework/CHIP/**",
+        "${workspaceFolder}/src/messaging/**",
+        "${workspaceFolder}/src/protocols/**",
+        "${workspaceFolder}/src/tracing/**",
+        "${workspaceFolder}/src/transport/**",
+        "${workspaceFolder}/src/inet/**",
+        "${workspaceFolder}/src/credentials/**",
+        "${workspaceFolder}/src/data_model/**",
```
Comment (bzbarsky-apple) [inline .vscode/settings.json:25]: We don't have an `src/data_model`?

File: .vscode/settings.json [hunk 2]
```diff
@@ -14,7 +14,18 @@
...
+        "${workspaceFolder}/src/app/**",
+        "${workspaceFolder}/src/crytpo/**",
```
Comment (bzbarsky-apple) [inline .vscode/settings.json:27]: "crypto"?

File: .vscode/settings.json (hunk 1 above)
Comment (woody-apple) [reply .vscode/settings.json:25]: https://github.com/project-chip/connectedhomeip/pull/33582

File: .vscode/settings.json (hunk 2 above)
Comment (woody-apple) [reply .vscode/settings.json:27]: https://github.com/project-chip/connectedhomeip/pull/33582

**Pull request merged**